### PR TITLE
Change balance attribute to is_balanced method

### LINKED ISSUE #29: Change the balance attribute on ChemicalEquation to a method
As suggested in a reply on the [Python Reddit](https://www.reddit.com/r/Python/comments/17xghha/comment/k9nw47j/?utm_source=share&utm_medium=web2x&context=3), the `balance` attribute on the `ChemicalEquation` class should be a method. The current implementation is:

```python
import chemics as cm
equation = cm.ChemicalEquation('2 HCl + 2 Na -> 2 NaCl + H2')
equation.balance
```

So using a method would be something like this:

```python
import chemics as cm
equation = cm.ChemicalEquation('2 HCl + 2 Na -> 2 NaCl + H2')
equation.is_balanced()
```

### PR DESCRIPTION
**Pull request by ghost**
Instead of a full refactor, I decided to remove the `balance` attribute from ChemicalEquation and convert the `_check_balance` method to the `is_balanced() -> bool` method.

Closes wigging/chemics#29

### CODE REVIEW

**Comment by wigging** [comment]
Please merge this into `wigging:dev` instead of `wigging:main`.

**Comment by sheodun** [comment]
Sorry, I have now requested to merge into `wigging:dev`

**Comment by wigging** [comment]
In the README you use `_is_balanced()` but it should be `is_balanced()`.

File: src/chemics/chemical_equation.py [hunk 1]
```diff
@@ -39,13 +39,11 @@ class ChemicalEquation:
...
     prod_mass : float
         Total mass of the products.
-    balance : bool
-        Balance between atomic elements in reactants and products.
 
     Examples
     --------
     >>> ce = cm.ChemicalEquation('2 HCl + 2 Na -> 2 NaCl + H2')
-    >>> ce.balance
+    >>> ce.is_balanced
```
Comment (wigging) [inline src/chemics/chemical_equation.py:46]: This should be `>>> ce.is_balanced()` otherwise the example in the docstring will fail.

**Review by wigging — CHANGES_REQUESTED** [review]
Please fix the docstring issue. Everything else looks fine.

File: src/chemics/chemical_equation.py (hunk 1 above)
Comment (sheodun) [reply src/chemics/chemical_equation.py]: Fixed.

**Review by wigging — APPROVED** [review]

**Pull request merged**